core/src/findlib/bfile.cc

```diff
@@ -1145,10 +1161,27 @@ ssize_t bwrite(BareosFilePacket* bfd, void* buf, size_t count)
     // plugin does read/write
     if (!bfd->do_io_in_core) { return plugin_bwrite(bfd, buf, count); }
 
+  const char* ptr = static_cast<const char*>(buf);
+
   Dmsg1(400, "bwrite handled in core via bfd->filedes=%d\n", bfd->filedes);
-  ssize_t status = write(bfd->filedes, buf, count);
+  ASSERT(static_cast<ssize_t>(count) >= 0);
+  ssize_t bytes_written = 0;
+  while (bytes_written < static_cast<ssize_t>(count)) {
```

Comment:
_bread_ is called in a loop; bwrite is often called just once, throwing an error if the write did not write everything (See e.g. `StoreData` in restore.cc).

I changed bread to match and also because often times having one big buffer is better for the consumer than multiple smaller ones (but its true that at least on linux, the read should almost always read the whole buffer since we are using pretty small ones).